core/codegen/src/lib.rs

```diff
@@ -774,6 +774,28 @@ pub fn derive_from_form(input: TokenStream) -> TokenStream {
     emit!(derive::from_form::derive_from_form(input))
 }
 
+/// Derive for the [`FromParam`] trait.
+///
+/// The [`FromParam`] derive can be applied to enums with nullary
+/// (zero-length) fields:
+///
+/// ```rust
+/// # #[macro_use] extern crate rocket;
+/// #
+/// #[derive(FromParam)]
+/// enum MyParam {
+///     A,
+///     B,
+/// }
```

Comment:
Can we make this a full doc-test?

Reply:
I am not sure what you mean by that? You mean ensuring that it does compile with `rustdoc --test`?

Reply:
I was thinking something more like
```rust
assert_eq!(MyParam::from_param("A").unwrap(), MyParam::A);
```
This can also be nice for documentation, since it shows what the parameter value is that can be parsed.